### PR TITLE
[hotfix] upgrade json-path and avro version

### LINKED ISSUE #2965: [Bug] upgrade to avro 1.11.3 due to cve
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

main branch

### Compute Engine

n/a

### Minimal reproduce step

n/a

### What doesn't meet your expectations?

Podlings should ideally use secure libs

### Anything else?

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-39410

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### LINKED ISSUE #2966: [Bug] upgrade to json-path 2.9.0
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

main branch

### Compute Engine

n/a

### Minimal reproduce step

n/a

### What doesn't meet your expectations?

podlings should ideally use secure libs

### Anything else?

CVE-2023-51074

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Linked issue: 
close #2966 
close #2965

### CODE REVIEW

File: paimon-filesystems/paimon-s3-impl/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -14,7 +14,6 @@ This project bundles the following dependencies under the Apache Software Licens
 - com.amazonaws:jmespath-java:1.12.319
 - com.fasterxml.jackson.core:jackson-annotations:2.14.2
 - com.fasterxml.jackson.core:jackson-core:2.14.2
-- com.fasterxml.jackson.core:jackson-databind:2.14.2
```
Comment (JingsongLi) [inline paimon-filesystems/paimon-s3-impl/src/main/resources/META-INF/NOTICE:17]: Why update here?

Comment (zhuangchong) [reply paimon-filesystems/paimon-s3-impl/src/main/resources/META-INF/NOTICE:17]: I made changes based on UT.

https://github.com/apache/incubator-paimon/actions/runs/8228264920/job/22497412307#step:5:80

image

Comment (zhuangchong) [reply paimon-filesystems/paimon-s3-impl/src/main/resources/META-INF/NOTICE]: Sorry, the notice checking UT is flawed, I'll check it later.

Comment (zhuangchong) [reply paimon-filesystems/paimon-s3-impl/src/main/resources/META-INF/NOTICE]: I have restored this file.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**